kong/db/declarative/init.lua

```diff
@@ -926,7 +926,43 @@ do
     local default_ws
     ok, err, default_ws = declarative.load_into_cache(entities, meta, hash, SHADOW)
     if ok then
-      ok, err = worker_events.post("declarative", "flip_config", default_ws)
+      local router_hash
+      local plugins_hash
+      local balancer_hash
+      if hashes then
+        if hashes.services ~= DECLARATIVE_EMPTY_CONFIG_HASH then
+          if hashes.routes ~= DECLARATIVE_EMPTY_CONFIG_HASH then
+            router_hash = md5(hashes.services .. hashes.routes)
+          else
+            router_hash = hashes.services
+          end
+        elseif hashes.routes ~= DECLARATIVE_EMPTY_CONFIG_HASH then
+          router_hash = hashes.routes
+        else
+          router_hash = DECLARATIVE_EMPTY_CONFIG_HASH
+        end
+
+        plugins_hash = hashes.plugins
+
+        if hashes.upstreams ~= DECLARATIVE_EMPTY_CONFIG_HASH then
+          if hashes.targets ~= DECLARATIVE_EMPTY_CONFIG_HASH then
+            balancer_hash = md5(hashes.upstreams .. hashes.targets)
```

Comment:
maybe we simply the logic here to always just do `md5(hashes.upstreams .. hashes.targets)`? as balancer_hash is changed either upstreams or targets
changed